src/Aspire.Hosting.Yarp/YarpResource.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Yarp;
+
+/// <summary>
+/// A resource that represents a YARP resource independent of the hosting model.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+public class YarpResource(string name) : ContainerResource(name)
+{
+    /// <summary>
+    /// File path of the config file for this YARP resource.
+    /// </summary>
+    public string? ConfigFilePath { get; set; }
```

Comment:
Lets make this internal for now. I suspect we are going to want to change the way we pull together this configuration file where you can have a baseline configuration file that you specify via `WithConfigurationFile` and then have other extension methods that layer in additional things that you want to do. That means this value would be painful for back compat (although we should ship in 9.3 as preview anyway).